internal/splunk/httprequest.go

```diff
@@ -29,6 +32,23 @@ func HandleHTTPCode(resp *http.Response) error {
 		resp.StatusCode,
 		http.StatusText(resp.StatusCode))
 
+	// Check if there is any error text returned by Splunk that we can append to the error message
+	if resp.Body != nil {
+		var jsonResponse map[string]any
+		bodyString, _ := io.ReadAll(resp.Body)
+		resp.Body = io.NopCloser(bytes.NewBuffer(bodyString))
+		unmarshalError := json.Unmarshal(bodyString, &jsonResponse)
+
+		if unmarshalError == nil {
+			responseErrorValue, ok := jsonResponse["text"]
```

Comment:
Is `"text"` defined in a spec? It would be great to move it to a const and add the link to the spec in a comment

Reply:
Best I could find is:
https://docs.splunk.com/Documentation/Splunk/9.1.3/RESTUM/RESTusing#Example_B:_JSON_response_format_example